assembly/src/parsers/field_ops.rs

```diff
@@ -373,6 +384,84 @@ pub fn parse_gte(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), Assemb
     Ok(())
 }
 
+// POWER OF TWO HELPER FUNCTIONS
+// ================================================================================================
+
+/// Extract the least significant bit of the top element iteratively and performs
+/// power of 2 operation on the individual bit. These individual powers
+/// are combined later to calculate the power of 2 on the top value of the stack.
+///
+/// The expected starting state of the stack (from the top) is: [a, ...].
+///
+/// After these operations, the stack state will be: [2^a, ...].
+fn aggregate_power_2(span_ops: &mut Vec<Operation>, unsafe_mode: bool) {
+    const MOST_SIGNIFICANT_BIT: u32 = 5;
+
+    // `safe` Mode
+    if !unsafe_mode {
+        // Checks if the top element of the stack is less than 64 or not.
+        span_ops.push(Operation::Dup0);
+        span_ops.push(Operation::Push(Felt::new(64)));
+        check_lt(span_ops);
+        span_ops.push(Operation::Assert);
+    }
+
+    span_ops.push(Operation::Push(Felt::new(2)));
+    span_ops.push(Operation::Swap);
+    span_ops.push(Operation::Dup1);
+    span_ops.push(Operation::U32div);
+    span_ops.push(Operation::Incr);
+    span_ops.push(Operation::Swap);
+
+    // Extract the least significant bit of the top value in the stack & calculate the power of 2
+    // for this bit. Eg. 1100111 (59) after 1st iteration will become 11001. The least significant
+    // bit is 1 & the power of 2 for this bit in this iteration would be
+    // (1 * 2 ^ ((2 ^ 2 - 1)) + 1  = 8. Similarily in the next iteration the power of two of the
+    // least significant bit will be 1 as the bit is zero.
+    for idx in 1..MOST_SIGNIFICANT_BIT {
+        let pow_two_at_exp = pow_of_two_at_bit(idx);
+
+        call_dup_opcode(span_ops, idx);
+        span_ops.push(Operation::U32div);
+        span_ops.push(Operation::Push(Felt::new(pow_two_at_exp - 1)));
+        span_ops.push(Operation::Mul);
+        span_ops.push(Operation::Incr);
+        span_ops.push(Operation::Swap);
+    }
+
+    // Pow of 2 at 2^5.
+    let pow_two_at_five = pow_of_two_at_bit(MOST_SIGNIFICANT_BIT);
+
+    span_ops.push(Operation::Push(Felt::new(pow_two_at_five - 1)));
+    span_ops.push(Operation::Mul);
+    span_ops.push(Operation::Incr);
+
+    // Aggregates all the individual power of 2 at each bit to calculate the final result
+    // of the power of two operation.
+    span_ops.extend_from_slice(&[Operation::Mul; 5]);
```

Comment:
minor nit: you could use `push_many()` extension method (defined in `core`) here.

Reply:
I missed this. Thanks!